hermes/probe/check_nil.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Check_nil implements the probe operation for checking if the NIL file is
+// present and reading verifying its contents.
+// Package probe implements the probe that Hermes uses to monitor
+// a storage system.
+
+package probe
+
+import (
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/cloudprober/logger"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+
+	m "github.com/googleinterns/step224-2020/hermes/probe/metrics"
+	sjpb "github.com/googleinterns/step224-2020/hermes/proto"
+)
+
+// TODO(evanSpendlove): Update comment here
+// deleteRandomFile deletes a random file in the target storage system bucket.
+// It then checks that the file has been deleted by trying to get the object.
+// Arguments:
+//	- ctx: pass the context so this probe can be cancelled if needed.
+//	- target: pass the target run information.
+//	- client: pass an initialised storage client for this target system.
+//	- logger: pass the logger associated with the probe calling this function.
+// Returns:
+//	- err:
+//		- [...].hermes_file_missing: the file to be deleted does not exist in Hermes' StateJournal.
+//		- [...].file_missing: the file to be deleted could not be found in the target bucket.
+//		- [...].bucket_missing: the target bucket on this target system was not found.
+//		- [...].probe_failed: there was an error during one of the API calls and the probe failed.
+//		- [...].deleted_file_found: the file was deleted but it is still found in the target bucket.
+//		- [...].list_bucket_failed: the listBucket operation failed when checking if the target file was deleted.
+// TODO(evanSpendlove): Add exit status to return values.
+func CheckNilFile(ctx context.Context, target *Target, client *stiface.Client, logger *logger.Logger) (m.ExitStatus, error) {
```

Comment:
This function is extremely long. I would recommend braking it up.

Reply:
I've split the function into several small functions.